nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -440,3 +441,131 @@ def _compress_mask(self, mask: Tensor) -> Tensor:
             mask = torch.einsum(ein_expression, mask, torch.ones(self.block_sparse_size).to(mask.device))
 
         return (mask != 0).type_as(mask)
+
+
+class TaskGenerator:
+    """
+    This class used to generate config list for pruner in each iteration.
+    """
+    def __init__(self, origin_model: Module, origin_masks: Dict[str, Dict[str, Tensor]] = {},
+                 origin_config_list: List[Dict] = [], log_dir: str = '.', keep_intermidiate_result: bool = False):
```

Comment:
what if I run two same pruning tasks at the same time?
Will these two tasks interfere with each other because of the same path? If so suggest adding a unique prefix

Reply:
I am not sure, do you mean run two task generators at the same time with the same `log_dir`? In this situation, they will interfere with each other, and fixing this problem should use the different `log_dir`.

And two tasks pruning the same model (load from the same model path) will not interfere with each other, because the model file will not be modified after saving, only will be deleted after no more needed.

Reply:
Is there an interface for user to specify a different `log_dir` in the `AutoCompressPruner` or `SimulatedAnnealingPruner`?

Reply:
add subfolder under `log_dir`, format like `2021-09-03-14-19-38-99331`.